packages/playground/blueprints/src/lib/steps/login.ts

```diff
@@ -21,44 +19,28 @@ export type LoginStep = {
 	 */
 	username?: string;
 	/**
-	 * The password to log in with. Defaults to 'password'.
+	 * @deprecated The password field is deprecated and will be removed in a future version.
+	 * Only the username field is required for user authentication.
 	 */
 	password?: string;
 };
 
 /**
  * Logs in to Playground.
- * Under the hood, this function submits the [`wp-login.php`](https://developer.wordpress.org/reference/files/wp-login.php/) [form](https://developer.wordpress.org/reference/functions/wp_login_form/)
- * just like a user would.
+ * Under the hood, this function sets the `PLAYGROUND_AUTOLOGIN_USERNAME` constant
+ * which is used to automatically log in the user in
+ * packages/playground/remote/src/lib/playground-mu-plugin/playground-includes/login_step.php
  */
 export const login: StepHandler<LoginStep> = async (
 	playground,
-	{ username = 'admin', password = 'password' } = {},
+	{ username = 'admin' } = {},
 	progress
 ) => {
 	progress?.tracker.setCaption(progress?.initialCaption || 'Logging in');
-	// Allow WordPress to set the cookies.
-	await playground.request({
-		url: '/wp-login.php',
-	});
+	await playground.defineConstant('PLAYGROUND_AUTOLOGIN_USERNAME', username);
 
-	const response = await playground.request({
-		url: '/wp-login.php',
-		method: 'POST',
-		body: {
-			log: username,
-			pwd: password,
-			rememberme: 'forever',
-		},
+	// Make a request to the wp-admin to ensure the user is logged after the blueprint runs.
+	await playground.request({
```

Comment:
@adamziel I couldn't find a good way to access the request handler.

[The example we have doesn't work](https://github.com/WordPress/wordpress-playground/blob/27dc0401e540c0a84f9fd031254714748815f79a/packages/php-wasm/universal/src/lib/php-request-handler.ts#L126-L153).

How can we make this easy?

Reply:
@bgrgicak I'd like to stop using the `request()` method and remove it from the Playground interface accessible in Step handlers. It only makes sense in the browser and isn't portable.

Reply:
In terms of mental models, creating a site with a Blueprint is distinct from accessing the site later on. We can use a Blueprint to create a zip artifact. We shouldn't generate auth cookies in the step handler. Let's only install the autoloading plugin without actually logging the users in.

Reply:
One thing I wonder about here is whether this would break using Playground to test logged-out behavior.

For example, if people wanted to test logged-in and logged-out views of the front end with Playground, how would this work?

Reply:
> One thing I wonder about here is whether this would break using Playground to test logged-out behavior.

It will work in the same way it does today. It you use `?login=no` you won't be logged in.